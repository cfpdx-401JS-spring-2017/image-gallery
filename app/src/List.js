import React, { Component } from 'react';
import PropTypes from 'prop-types';

class List extends Component {

  static propTypes = {
    data: PropTypes.array.isRequired
  }

  render() {
    const { data } = this.props;

    return (
      <div>
        {data.map((img, i) => (
          <ul key={i}>
            <li>
              {img.title}
            </li>
            <li>
              <a href=''>
                {img.url}
              </a>
            </li>
            <li>
              {img.description}
            </li>
          </ul>
        ))}
      </div>
    );
  }
}

export default List;