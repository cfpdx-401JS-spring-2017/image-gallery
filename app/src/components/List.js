import React, { Component } from 'react';
import PropTypes from 'prop-types';

class List extends Component {

  static propTypes = {
    data: PropTypes.array.isRequired
  }

  render() {
    const { data, onDelete } = this.props;

    return (
      <div>
        {data.map((img) => (
          <ul key={img._id}>
            <li className="list-title">
              {img.title}
            </li>
            <li>
              <a href="">
                {img.url}
              </a>
            </li>
            <li>
              {img.description}
            </li>
            <li>
              <button onClick={() => onDelete(img._id)}
                      className="delete-button"
              >
                Delete ðŸ—‘
              </button>
            </li>
          </ul>
        ))}
      </div>
    );
  }
}

export default List;