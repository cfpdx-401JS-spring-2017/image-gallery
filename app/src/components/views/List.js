import React from 'react';
import PropTypes from 'prop-types';
import ListImage from './ListImage';

List.propTypes = {
  poms: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    url: PropTypes.string.isRequired
  }))
};

export default function List({ poms, handleDelete }) {
  return (
    <ul>
      {poms.map((pom, i) => <ListImage
        key={pom._id}
        {...pom}
        onDelete={() => handleDelete(pom._id, i)}
      />)}
    </ul>
  );
}