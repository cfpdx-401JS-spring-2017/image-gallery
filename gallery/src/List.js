import React, { Component } from 'react';

class List extends Component {
  constructor() {
    super();
    this.state ={

    };
  }
  

  render() {
    return (
      <div className="List">
        {this.props.images.map((image, i) =>{
            let {title, description, url} = image;
          return (
            <div key={i} style={{ width: '30vw', margin: '5px 5px'}}>
              <h3>{title}</h3> 
              <p>{description}</p> 
              <p>{url}</p>
            </div>
          );
          
        })}
      </div>  
    );
  }
}

export default List;