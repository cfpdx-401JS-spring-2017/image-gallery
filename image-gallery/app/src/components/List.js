import React from 'react';
import PropTypes from 'prop-types';
import DeleteImage from './DeleteImage'


    export default function List (props) {

        return (
            <div>
                {props.images.map(image => (
                    <ul className="list-view" key={image._id}>
                        <li className="image-caption">{image.title} </li>
                        <li className="image-caption">{image.description} </li>
                        <li className="image-caption">{image.url} </li>
                        <li><DeleteImage removeImage={() => props.removeImage(props.albumId, image._id)}/></li>
                    </ul>))}
            </div>


        )
    }

    List.proptypes = {
images:PropTypes.array.isRequired,
onDelete:PropTypes.func.isRequired

};

