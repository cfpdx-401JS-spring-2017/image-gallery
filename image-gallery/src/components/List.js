import React from 'react';
import PropTypes from 'prop-types';
import DeleteImage from './DeleteImage'


    export default function List (props) {
  
        return (
            <div>
                {props.images.map(image => (
                    <ul className="list-view" key={image.url}>
                        <li className="image-caption">{image.title} </li>
                        <li className="image-caption">{image.description} </li>
                        <li className="image-caption">{image.url} </li>
                        <li><DeleteImage onClick={() => this.props.onDelete(image._id)}/></li>
                    </ul>))}
                    <DeleteImage/>
            </div>


        )
    }

    List.proptypes = {
images:PropTypes.array.isRequired
};

