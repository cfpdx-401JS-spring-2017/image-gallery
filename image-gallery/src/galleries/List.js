import React, { Component } from 'react';



export default class List extends Component {
  
    render() {
        return (
            <div>
                {this.props.images.map(image => (
                    <ul key={image.url}>
                        <li >{image.title} </li>
                        <li>{image.description} </li>
                        <li>{image.url} </li>
                    </ul>))}
            </div>


        )
    }


}