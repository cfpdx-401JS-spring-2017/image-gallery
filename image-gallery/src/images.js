import lordsOfDogtown from './photos/lordsOfDogtown.jpg';
import jayAdams from './photos/jayAdams.png';
import peggyOki from './photos/peggyOki.jpg';
import mikeKolar from './photos/mikeKolar.jpg';
import peggyTurner from './photos/peggyTurner.jpg';
import lauraThornhill from './photos/lauraThornhill2.jpg';
import rampage from './photos/rampage.jpg';
import shortId from 'shortid';
// let shortId = require('shortid');

const images = [
    {
        title: 'Z Boys',
        description: 'Zephyr skate team 1970\'s',
        url: lordsOfDogtown,
        _id: shortId.generate()
    },

    {
        title: 'Turn Only',
        description: 'Jay Adams gets low',
        url: jayAdams,
        _id: shortId.generate()
    },

    {
        title: 'Peggy Does Venice',
        description: 'Peggy Oki repping Zephr',
        url: peggyOki,
        _id: shortId.generate()
    },

    {
        title: 'Jumping Homies',
        description: 'Mike Kolar 1970\'s',
        url: mikeKolar,
        _id: shortId.generate()
    },

    {
        title: "Tippin' Toes",
        description: 'Laura Thornhill in skate competition 1970\'s',
        url: lauraThornhill,
        _id: shortId.generate()
    },

    {
        title: "Van Fans",
        description: 'Peggy Turner repping Vans in skate competition',
        url: peggyTurner,
        _id: shortId.generate()
    },

    {
        title: "Jumping Homies 2.0",
        description: 'Rampage Expo 1970\'s',
        url: rampage,
        _id: shortId.generate()
    }

];

export default {
    get() { return Promise.resolve(images) }
}