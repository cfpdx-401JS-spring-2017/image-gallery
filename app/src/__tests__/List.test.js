import React from 'react';
import renderer from 'react-test-renderer';
import List from '../components/List';
import images from './_data';

describe('List', () => {
  it('renders a list', () => {
    const tree = renderer.create(<List data={images}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});