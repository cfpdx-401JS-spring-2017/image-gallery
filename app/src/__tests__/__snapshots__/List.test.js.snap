// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders a list 1`] = `
<div>
  <ul>
    <li
      className="list-title"
    >
      Black & Tan
    </li>
    <li>
      <a
        href=""
      >
        http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg
      </a>
    </li>
    <li>
      Seriously heart-melting black and tan bunny... OMG &lt;3
    </li>
    <li>
      <button
        className="delete-button"
        onClick={[Function]}
      >
        Delete ðŸ—‘
      </button>
    </li>
  </ul>
  <ul>
    <li
      className="list-title"
    >
      Falling Bunny
    </li>
    <li>
      <a
        href=""
      >
        http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg
      </a>
    </li>
    <li>
      I don't know if this bunny fell and the picture was taken on impact, but I certainly prefer to think of it that way
    </li>
    <li>
      <button
        className="delete-button"
        onClick={[Function]}
      >
        Delete ðŸ—‘
      </button>
    </li>
  </ul>
  <ul>
    <li
      className="list-title"
    >
      Marshmellow Puff
    </li>
    <li>
      <a
        href=""
      >
        http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg
      </a>
    </li>
    <li>
      Is it a rodent or a rabbit? I had to look it up to be sure. Then again, I'm still not sure
    </li>
    <li>
      <button
        className="delete-button"
        onClick={[Function]}
      >
        Delete ðŸ—‘
      </button>
    </li>
  </ul>
</div>
`;
